src/Rest/Routes/AbstractOauth.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+/**
+ * The class register route for Oauth.
+ *
+ * @package EightshiftForms\Rest\Routes
+ */
+
+declare(strict_types=1);
+
+namespace EightshiftForms\Rest\Routes;
+
+use EightshiftForms\Exception\UnverifiedRequestException;
+use EightshiftFormsVendor\EightshiftFormsUtils\Helpers\UtilsGeneralHelper;
+use EightshiftFormsVendor\EightshiftFormsUtils\Helpers\UtilsSettingsHelper;
+use EightshiftFormsVendor\EightshiftFormsUtils\Rest\Routes\AbstractUtilsBaseRoute;
+use WP_REST_Request;
+
+/**
+ * Class AbstractOauth
+ */
+abstract class AbstractOauth extends AbstractUtilsBaseRoute
+{
+	/**
+	 * Dynamic name route prefix for oauth.
+	 *
+	 * @var string
+	 */
+	public const ROUTE_PREFIX_OAUTH_API = 'oauth';
+
+	/**
+	 * Method that returns rest response
+	 *
+	 * @param WP_REST_Request $request Data got from endpoint url.
+	 *
+	 * @throws UnverifiedRequestException Wrong config error.
+	 *
+	 * @return WP_REST_Response|mixed If response generated an error, WP_Error, if response
+	 *                                is already an instance, WP_HTTP_Response, otherwise
+	 *                                returns a new WP_REST_Response instance.
+	 */
+	public function routeCallback(WP_REST_Request $request)
+	{
+		if (!$this->checkEnabledConnectionPermission()) {
+			$this->redirect(\esc_html__('You do not have permission to access this page.', 'eightshift-forms'));
+		}
+
+		try {
+			$code = $request->get_param('code');
+
+			// No code parameter? Redirect to login.
+			if (!$code) {
+				$this->redirect(\esc_html__('Code parameter is empty.', 'eightshift-forms'));
+			}
+
+			// If code parameter is not a string, redirect to login.
+			if (!\is_string($code)) {
+				$this->redirect(
+					\sprintf(
+						/* translators: % denotes type or code parameter. */
+						\esc_html__('Code parameter should be a string, %s provided.', 'eightshift-forms'),
+						\gettype($code)
+					)
+				);
+			}
+
+			// Sanitize code parameter for security.
+			$code = \sanitize_text_field($code);
+
+			return $this->submitAction($code);
+		} catch (UnverifiedRequestException $e) {
+			$this->redirect(\esc_html__('Error.', 'eightshift-forms'));
+		}
+	}
+
+	/**
+	 * Implement submit action.
+	 *
+	 * @param string $code The code.
+	 *
+	 * @return mixed
+	 */
+	abstract protected function submitAction(string $code);
```

Comment:
```suggestion
	abstract protected function submitAction(string $code): mixed;
```
Given this can be anything is there a point to hinting at all? 😄

Reply:
no need for the mixed